src/features/usersActionBar/SearchInput.tsx

```diff
@@ -2,38 +2,76 @@ import React, { FC } from 'react';
 import { useAppSelector, useAppDispatch } from '../../app/hooks';
 import {
   selectUsers,
-  selectSearchText,
+  selectSearchNameText,
+  selectSearchLoginText,
+  selectSearchEmailText,
+  selectSearchPhoneText,
+  selectSearchRatingText,
   addSearchText,
   loadFilteredUsers,
+  selectSearchText,
 } from '../users/usersSlice';
+import { SortItem } from '../../types/types';
 
-const SearchInput: FC = () => {
+const SearchInput: FC<SortItem> = ({ columnName }: SortItem) => {
   const dispatch = useAppDispatch();
   const users = useAppSelector(selectUsers);
   const searchText = useAppSelector(selectSearchText);
+  let inputValue;
+
+  switch (columnName) {
+    case 'name': {
+      inputValue = useAppSelector(selectSearchNameText);
+      break;
+    }
+    case 'login': {
+      inputValue = useAppSelector(selectSearchLoginText);
+      break;
+    }
+    case 'email': {
+      inputValue = useAppSelector(selectSearchEmailText);
+      break;
+    }
+    case 'phone': {
+      inputValue = useAppSelector(selectSearchPhoneText);
+      break;
+    }
+    case 'rating': {
+      inputValue = useAppSelector(selectSearchRatingText);
+      break;
+    }
+    default:
+  }
 
   const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
     event.preventDefault();
     const value = event.target.value.trim();
-    dispatch(addSearchText(value));
+
+    dispatch(addSearchText({
+      ...searchText,
+      [columnName]: value,
+    }));
+
     if (value) {
-      const filteredUsers = users.filter((user) => (
-        user.name.indexOf(value) !== -1
-        || user.login.indexOf(value) !== -1
-        || user.email.indexOf(value) !== -1
-        || user.phone.indexOf(value) !== -1
-        || user.rating === Number(value)
-      ));
+      const filteredUsers = users.filter((user) => {
+        if (columnName === 'rating') {
+          return user[columnName] === value;
+        }
+        return (user[columnName] as string).toLowerCase().indexOf(value) !== -1;
+      });
       dispatch(loadFilteredUsers(filteredUsers));
     } else {
       dispatch(loadFilteredUsers(users));
     }
   };
 
   return (
-    <form>
-      <input placeholder="Type the search text" onChange={handleChange} value={searchText} />
-    </form>
+    <div>
+      <span>
+        {`Search in ${columnName} column`}
```

Comment:
Наверное, можно чуть-чуть проще:
```suggestion
        Search in {{columnName}} column
```